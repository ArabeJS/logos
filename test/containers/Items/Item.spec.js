import React from 'react';
import { mount } from 'enzyme';

import Item from 'containers/Items/Item';

const mockHandleClickTag = jest.fn();

function setup() {
  const props = {
    data: {
      name: 'React',
      files: [
        'react.svg',
      ],
      shortname: 'react',
      url: 'https://facebook.github.io/react/',
      tags: [
        'front-end framework',
        'javascript',
        'facebook',
      ],
      categories: [
        'developers',
      ],
      updated: '2015-06-11',
    },
    index: 0,
    handleClickTag: mockHandleClickTag,
  };

  return mount(<Item {...props} />);
}

describe('Item', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

