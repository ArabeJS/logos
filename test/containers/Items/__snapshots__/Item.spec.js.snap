// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render properly 1`] = `"<li><a href=\\"https://facebook.github.io/react/\\" target=\\"_blank\\" class=\\"app__images__img\\" data-shortname=\\"react\\"><img src=\\"https://cdn.svgporn.com/logos/react.svg\\" alt=\\"React\\" class=\\"react\\"></a><div class=\\"app__images__info\\"><h5><a href=\\"https://facebook.github.io/react/\\" target=\\"_blank\\">React</a></h5><div class=\\"app__images__tags\\"><a href=\\"#tag\\" data-name=\\"front-end framework\\">#front-end framework</a><a href=\\"#tag\\" data-name=\\"javascript\\">#javascript</a><a href=\\"#tag\\" data-name=\\"facebook\\">#facebook</a></div><div class=\\"app__images__download\\"><a href=\\"https://cdn.svgporn.com/logos/react.svg\\" class=\\"btn btn-sm btn-outline-secondary btn-icon\\" download=\\"\\"><i class=\\"i-download\\"></i><span>download</span></a></div></div></li>"`;
